api/src/main/java/com/velocitypowered/api/proxy/server/QueryResponse.java

```diff
@@ -0,0 +1,306 @@
+package com.velocitypowered.api.proxy.server;
+
+import com.google.common.base.Preconditions;
+import com.velocitypowered.api.proxy.config.ProxyConfig;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * GS4 query response. This class is immutable.
+ */
+public final class QueryResponse {
+    private final String hostname;
+    private final String gameVersion;
+    private final String map;
+    private final int currentPlayers;
+    private final int maxPlayers;
+    private final String proxyHost;
+    private final int proxyPort;
+    private final Collection<String> players;
+    private final String proxyVersion;
+    private final List<PluginInformation> plugins;
+
+    public QueryResponse(String hostname, String gameVersion, String map, int currentPlayers, int maxPlayers, String proxyHost, int proxyPort, Collection<String> players, String proxyVersion, List<PluginInformation> plugins) {
```

Comment:
Hide the constructor. It's so long that it's more idiomatic to use the builder in virtually all cases.

Reply:
By hiding you mean...?

Reply:
Set visibility to `private` so that only the builder can access the constructor.

Reply:
Ah, indeed.